### PR TITLE
replace anonymous runnable class with lambda

### PR DESCRIPTION
**Pull request by elsazac**

### CODE REVIEW

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java [hunk 1]
```diff
@@ -872,18 +872,14 @@ public void processEvent(VerifyEvent e) {
 	private void filterProposals() {
 		++ fInvocationCounter;
 		Control control= fViewer.getTextWidget();
-		control.getDisplay().asyncExec(new Runnable() {
+		control.getDisplay().asyncExec(() -> {
```
Comment (mickaelistria) [inline bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java:876]: This is not equivalent and will most likely cause bugs:

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java [hunk 2]
```diff
@@ -872,18 +872,14 @@ public void processEvent(VerifyEvent e) {
...
 			long fCounter= fInvocationCounter;
```
Comment (mickaelistria) [inline bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java]: Before, fCounter is initialized just once, when `filterProposals` is invoked and the `Runnable` is created, ie before actually running the Runnable. While with the change, the value is set at execution of the Runnable, leading to...

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java [hunk 3]
```diff
@@ -872,18 +872,14 @@ public void processEvent(VerifyEvent e) {
...
-			@Override
-			public void run() {
-
-				if (fCounter != fInvocationCounter)
```
Comment (mickaelistria) [inline bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java:880]: This condition not being always true (before) but being always true (after)

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/ContextInformationPopup.java [hunk 4]
```diff
@@ -876,30 +876,24 @@ private void validateContextInformation() {
 		 */
 		if (!isValid(fContextInfoPopup))
 			return;
-
-		fContextInfoPopup.getDisplay().asyncExec(new Runnable() {
-
-			private ContextFrame fFrame= fContextFrameStack.peek();
```
Comment (mickaelistria) [inline bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/ContextInformationPopup.java]: A similar pattern here: the frame is set once at instantiation for some reason, not on the execution. This looks like an actual intent to capture the "initial value before execution".

**Review by mickaelistria — CHANGES_REQUESTED** [review]
I don't think this change is safe. Usually, a class that defines a field shouldn't be turned into a lambda as the field lifecycle (which value they get and where) matters.

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java (hunk 2 above)
Comment (elsazac) [reply bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java]: will moving `fCounter`  assignment outside of the lambda solve the problem?

Comment (mickaelistria) [reply bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java]: Yes, I think it would.

Comment (elsazac) [reply bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java]: @mickaelistria I have made that change accordingly and removed the problematic lambda. please have a look.

**Review by mickaelistria — APPROVED** [review]

**Comment by mickaelistria** [comment]
Thank you!

**Pull request merged**